### PR TITLE
Improve loop JitDump output

### PR DESCRIPTION
**Pull request by BruceForstall**
In addition,
- Added COMPlus_JitDumpFgBlockFlags to include BasicBlock flags in the JitDumpFg ".dot" file
- Added COMPlus_JitDumpFgLoopFlags to include loop table flags in the JitDumpFg ".dot" file
- Added cLoopPtr/dLoopPtr, cLoops/dLoops debugger-used functions, and changed cLoop to call the standard dumper
- Added clrjit.natvis debugger display for LoopDsc
- Improved a few comments

No asm diffs

### CODE REVIEW

**Comment by BruceForstall** [comment]
Here's a sample fragment of loop table output:

```
L00, from BB02 to BB02 (Head=BB01, Entry=BB02, Exit=BB02) [over V04 (ADD 1) from 0 LE 511]
L01, from BB04 to BB08 (Head=BB03, Entry=BB04, Exit=BB08), child loop = L02 [over V01 (ADD 1) from 1 LE 5]
L02, from BB05 to BB07 (Head=BB04, Entry=BB05, Exit=BB07, parent=L01), child loop = L03 [over V02 (ADD 1) from 1 LE 5]
L03, from BB06 to BB06 (Head=BB05, Entry=BB06, Exit=BB06, parent=L02) [over V03 (ADD 1) from 1 LE 5]
```

**Comment by BruceForstall** [comment]
@AndyAyersMS @dotnet/jit-contrib PTAL

**Review by AndyAyersMS — APPROVED** [review]
Looks good.

Pointed out a few tiny things you might consider changing.

File: src/coreclr/jit/fgbasic.cpp [hunk 1]
```diff
@@ -4701,7 +4701,7 @@ void Compiler::fgRemoveBlock(BasicBlock* block, bool unreachable)
             {
                 loopAlignCandidates++;
                 succBlock->bbFlags |= BBF_LOOP_ALIGN;
-                JITDUMP("Propagating LOOP_ALIGN flag from " FMT_BB " to " FMT_BB " for loop# %d.", block->bbNum,
+                JITDUMP("Propagating LOOP_ALIGN flag from " FMT_BB " to " FMT_BB " for loop# %d.\n", block->bbNum,
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply src/coreclr/jit/fgbasic.cpp:4704]: Will fix in a follow-up

File: src/coreclr/jit/loopcloning.cpp [hunk 2]
```diff
@@ -1604,7 +1604,7 @@ bool Compiler::optIsLoopClonable(unsigned loopInd)
 
     if (end->bbJumpDest != beg)
     {
-        JITDUMP("Loop cloning: rejecting loop " FMT_LP ". Branch at loop 'end' not looping to 'begin'.\n", loopInd);
+        JITDUMP("Loop cloning: rejecting loop " FMT_LP ". Branch at loop 'bottom' not looping to 'top'.\n", loopInd);
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply src/coreclr/jit/loopcloning.cpp:1607]: It turns out I already fixed this in a follow-up

File: src/coreclr/jit/optimizer.cpp [hunk 3]
```diff
@@ -9122,3 +9267,67 @@ void Compiler::optRemoveRedundantZeroInits()
...
+//
+// Arguments:
+//      loopNum - the loop number to check
+//
+bool Compiler::optAnyChildNotRemoved(unsigned loopNum)
+{
+    assert(loopNum < optLoopCount);
+
+    // Now recursively mark the children.
+    for (BasicBlock::loopNumber l = optLoopTable[loopNum].lpChild; //
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply src/coreclr/jit/optimizer.cpp:9286]: I haven't yet introduced any iterators for the loop structures. Maybe someday...?

**Pull request merged**